### PR TITLE
Swiftier code

### PR DESCRIPTION
**Pull request by Mongey**
* Moves networking code out of the models and into `CircleCIClient.swift`
* Adds tests for decoding + filtering

### CODE REVIEW

**Comment by olets** [comment]
Wow. You're crushing it 👏

File: SeaEye/controllers/SeaEyeIconController.swift [hunk 1]
```diff
@@ -33,32 +33,29 @@ class SeaEyeIconController: NSViewController {
...
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "SeaEyeAlert"),
+                                               object: nil,
+                                               queue: OperationQueue.main,
+                                               using: alert)
+        
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "SeaEyeRedBuild"),
+                                               object: nil,
+                                               queue: OperationQueue.main,
+                                               using: setRedBuildIcon)
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "SeaEyeRedBuild"),
```
Comment (nolaneo) [inline SeaEye/controllers/SeaEyeIconController.swift]: `SeaEyeGreenBuild`?

File: SeaEye/models/CircleCIModel.swift [hunk 2]
```diff
@@ -47,53 +50,40 @@ class CircleCIModel: NSObject {
...
                         default: break;
                     }
                 }
             }
             
-            if failures > 1 {
-                print("Has multiple failues")
-                let info = ["build": failedBuild, "count": failures] as [String : Any]
+            if failures > 0 {
+//                print("Has red build \(failedBuild!.subject)")
```
Comment (nolaneo) [inline SeaEye/models/CircleCIModel.swift]: ⚡️

**Comment by Mongey** [comment]
Thanks

**Pull request merged**